packages/react-vapor/src/components/refresh/RefreshCallback.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {RefreshCallBackActions} from './RefeshCallbackActions';
+import {RefreshStatus, RefreshStatusSelectors} from './RefreshCallbackReducer';
+
+export interface IRefreshCallbackOwnProps {
+    id: string;
+    time?: number;
+    incremental?: number;
+    intervalInMs?: number;
+    callback: (start: () => void) => void;
+    renderCount?: (count: number) => React.ReactNode;
+}
+
+export interface IRefreshCallbackState {
+    count: number;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IRefreshCallbackOwnProps) => ({
+    status: RefreshStatusSelectors.getRefreshStatus(state, {id: ownProps.id}),
+});
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IRefreshCallbackOwnProps) => ({
+    start: () => dispatch(RefreshCallBackActions.start(ownProps.id)),
+    stop: () => dispatch(RefreshCallBackActions.stop(ownProps.id)),
+    inProgress: () => dispatch(RefreshCallBackActions.inProgress(ownProps.id)),
+});
+
+export interface IRefreshCallbackProps
+    extends IRefreshCallbackOwnProps,
+        Partial<ReturnType<typeof mapStateToProps>>,
+        Partial<ReturnType<typeof mapDispatchToProps>> {}
+
+class RefreshCallbackDisconnected extends React.PureComponent<IRefreshCallbackProps, IRefreshCallbackState> {
+    static defaultProps: Partial<IRefreshCallbackProps> = {
+        time: 10,
+        incremental: 1,
+        intervalInMs: 1000,
+        renderCount: (count: number) => <span>Auto refresh in {count} seconds</span>,
+    };
+
+    private activeInterval: any;
+
+    constructor(props: IRefreshCallbackProps) {
+        super(props);
+        this.state = {
+            count: this.props.time,
+        };
+    }
+
+    private startInterval() {
+        this.activeInterval = setInterval(() => {
+            this.setState((prevState: IRefreshCallbackState) => ({
+                count: prevState.count - this.props.incremental,
+            }));
+        }, this.props.intervalInMs);
+        this.props.inProgress();
+    }
+
+    componentDidMount() {
+        this.startInterval();
+    }
+
+    componentDidUpdate(prevProps: Readonly<IRefreshCallbackProps>, prevState: Readonly<IRefreshCallbackState>) {
+        if (this.props.status === RefreshStatus.start) {
```

Comment:
Just a suggestion, I think it would be nice to have getters for those comparisons

```
private get isStarted() {
   return this.props.status === RefreshStatus.start;
}
...
if (this.isStarted) {

}
```